curve25519-dalek/src/backend/serial/u64/field.rs

```diff
@@ -172,16 +198,28 @@ impl<'a> Add<&'a FieldElement51> for &FieldElement51 {
         output += _rhs;
         */
         /* MODIFIED CODE */
-        for i in 0..5 {
-            proof {
-                assume(false);
-            }
+        let ghost original_limbs = self.limbs;
+        for i in 0..5
+            invariant
+                forall|j: int|
+                    #![auto]
+                    0 <= j < i ==> output.limbs[j] == original_limbs[j] + _rhs.limbs[j],
+                forall|j: int| #![auto] i <= j < 5 ==> output.limbs[j] == original_limbs[j],
+                forall|j: int|
+                    0 <= j < 5 ==> #[trigger] original_limbs[j] + _rhs.limbs[j] <= u64::MAX,
+        {
+            assert(output.limbs[i as int] == original_limbs[i as int]);
+            assert(original_limbs[i as int] + _rhs.limbs[i as int] <= u64::MAX);
```

Comment:
```suggestion
```
Are these asserts necessary? if it's to trigger the `add_req` forall, I'd add a comment that that's what they do.